lib/connectors_sdk/base/http_call_wrapper.rb

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'bson'
+
+module ConnectorsSdk
+  module Base
+    class HttpCallWrapper
+      def extractor(params)
+        extractor_class.new(
+          content_source_id: params[:content_source_id] || "GENERATED-#{BSON::ObjectId.new}",
+          service_type: service_type,
+          authorization_data_proc: proc { { access_token: params[:access_token] } },
+          client_proc: proc { client(params) },
+          config: config(params),
+          features: params.fetch(:features, {}) || {}
+        )
+      end
+
+      def document_batch(params)
+        results = []
+
+        extractor = extractor(params)
+
+        extractor.yield_document_changes(:break_after_page => true, :modified_since => extractor.config.cursors['modified_since']) do |action, doc, download_args_and_proc|
+          download_obj = nil
+          if download_args_and_proc
+            download_obj = {
+              id: download_args_and_proc[0],
+              name: download_args_and_proc[1],
+              size: download_args_and_proc[2],
+              download_args: download_args_and_proc[3]
+            }
+          end
+
+          results << {
+            :action => action,
+            :document => doc,
+            :download => download_obj
+          }
+        end
+
+        [results, extractor.config.cursors, extractor.completed]
+      rescue custom_client_error => e
+        raise e.status_code == 401 ? ConnectorsShared::InvalidTokenError : e
```

Comment:
I realize that you are catching `ConnectorsShared::InvalidTokenError` everywhere. and always treating them the same way by converting them to a 401. I am wondering if this could be done in a single place in this class, that would decorate all functions

Reply:
Yes we can do this. Will update soon